src/seismometer/table/analytics_table.py

```diff
@@ -0,0 +1,838 @@
+                    ),
+                )
+        return gt
+
+    def add_coloring_parity(self, gt, columns=None, even_color="#F2F2F2", odd_color="white"):
+        """
+        Adds alternating row colors to the specified columns in the table.
+
+        Parameters
+        ----------
+        gt : GT
+            The table object to which the alternating row colors will be added.
+        columns : Optional[List[str]], optional
+            The list of columns to which the alternating row colors will be applied, by default None.
+            If None, all columns are considered.
+        even_color : str, optional
+            The color for even rows, by default "#F2F2F2" (light gray).
+        odd_color : str, optional
+            The color for odd rows, by default "white".
+
+        Returns
+        -------
+        gt : GT
+            The table object with alternating row colors.
+        """
+        gt = gt.tab_style(
+            style=style.fill(color=even_color),
+            locations=loc.body(
+                columns=columns,
+                rows=[row for row in range(self.num_of_rows) if (row % self.rows_group_length) % 2 == 0],
+            ),
+        ).tab_style(
+            style=style.fill(color=odd_color),
+            locations=loc.body(
+                columns=columns,
+                rows=[row for row in range(self.num_of_rows) if (row % self.rows_group_length) % 2 == 1],
+            ),
+        )
+        return gt
+
+    def group_columns_by_metric_value(self, gt, columns, value):
+        gt = (
+            gt.tab_spanner(label=f"{self.metric}={value}", columns=columns)
+            .cols_label(**{col: "_".join(col.split("_")[1:]) for col in columns})
+            .tab_style(
+                style=style.borders(sides=["left"], weight="2px", color="black"),
+                locations=loc.body(columns=[columns[0]]),
+            )
+            .tab_style(
+                style=style.borders(sides=["right"], weight="2px", color="black"),
+                locations=loc.body(columns=[columns[-1]]),
+            )
+        )
+        gt = self.color_group_of_columns(gt, columns)
+
+        return gt
+
+    def color_group_of_columns(self, gt, columns):
+        if not self.spanner_colors:
+            return gt
+        elif self.alternating_row_colors:
+            gt = self.add_coloring_parity(
+                gt,
+                columns,
+                even_color=lighten_color(self.spanner_colors[self.spanner_color_index]),
+                odd_color=self.spanner_colors[self.spanner_color_index],
+            )
+        else:
+            gt = gt.tab_style(
+                style=style.fill(color=self.spanner_colors[self.spanner_color_index]),
+                locations=loc.body(columns=columns),
+            )
+        # increase the spanner_color_index after using spanner colors.
+        self.spanner_color_index = (self.spanner_color_index + 1) % len(self.spanner_colors)
```

Comment:
another good wrapper function that only depends on the table itself.

Reply:
Removed also, as we are not coloring grouped columns.